base/applications/cleanmgr/util.c

```diff
@@ -0,0 +1,1120 @@
+            {
+                return;
+            }
+
+            ZeroMemory(&wcv.DriveLetter, sizeof(wcv.DriveLetter));
+        }
+        SingleDrive += wcslen(SingleDrive) + 1;
+    }
+}
+
+void InitStartDlg(HWND hwnd, HBITMAP hBitmap)
+{
+    DWORD DwIndex = 0;
+    DWORD NumOfDrives = 0;
+    HWND hComboCtrl = NULL;
+    WCHAR *GatheredDriveLatter = NULL;
+    WCHAR LogicalDrives[ARR_MAX_SIZE] = { 0 };
+    WCHAR StringComboBox[ARR_MAX_SIZE] = { 0 };
+
+    hComboCtrl = GetDlgItem(hwnd, IDC_DRIVE);
+
+    if (hComboCtrl == NULL)
+    {
+        MessageBoxW(NULL, L"GetDlgItem() failed!", L"Error", MB_OK | MB_ICONERROR);
+        EndDialog(hwnd, IDCANCEL);
+        return;
+    }
+  
+    NumOfDrives = GetLogicalDriveStringsW(sizeof(LogicalDrives), LogicalDrives);
+    if (NumOfDrives == 0)
+    {
+        MessageBoxW(NULL, L"GetLogicalDriveStringsW() failed!", L"Error", MB_OK | MB_ICONERROR);
+        EndDialog(hwnd, IDCANCEL);
+        return;
+    }
+
+    if (NumOfDrives <= sizeof(LogicalDrives))
+    {
+        WCHAR* SingleDrive = LogicalDrives;
+        WCHAR RealDrive[ARR_MAX_SIZE] = { 0 };
+        WCHAR VolumeName[ARR_MAX_SIZE] = { 0 };
+        WCHAR StringComboBox[ARR_MAX_SIZE] = { 0 };
+        while (*SingleDrive)
+        {
+            if (GetDriveTypeW(SingleDrive) == ONLY_PHYSICAL_DRIVE)
+            {
+                StringCchCopyW(RealDrive, sizeof(RealDrive), SingleDrive);
+                RealDrive[wcslen(RealDrive) - 1] = '\0';
+                if (!GetVolumeInformationW(SingleDrive,
+                                      VolumeName,
+                                      ARR_MAX_SIZE,
+                                      NULL,
+                                      NULL,
+                                      NULL,
+                                      NULL,
+                                      0))
+                {
+                    MessageBoxW(NULL, L"GetVolumeInformationW() failed!", L"Error", MB_OK | MB_ICONERROR);
+                    EndDialog(hwnd, IDCANCEL);
+                    return;
+                }
+
+                StringCchPrintfW(StringComboBox, sizeof(StringComboBox), L"%s (%s)", VolumeName, RealDrive);
+                DwIndex = SendMessageW(hComboCtrl, CB_ADDSTRING, 0, (LPARAM)StringComboBox);
+                if (SendMessageW(hComboCtrl, CB_SETITEMDATA, DwIndex, (LPARAM)hBitmap) == CB_ERR)
+                {
+                    MessageBoxW(NULL, L"SendMessageW() failed!", L"Error", MB_OK | MB_ICONERROR);
+                    EndDialog(hwnd, IDCANCEL);
+                    return;
+                }
+            }
+            SingleDrive += wcslen(SingleDrive) + 1;
+        }
+    }
+   
+    ComboBox_SetCurSel(hComboCtrl, 0);
+    SendMessageW(hComboCtrl, CB_GETLBTEXT, (WPARAM)0, (LPARAM)StringComboBox);
+    StringComboBox[wcslen(StringComboBox) - 1] = L'\0';
+    GatheredDriveLatter = wcsrchr(StringComboBox, L':');
+    GatheredDriveLatter--;
```

Comment:
???

Reply:
@learn-more This part of code is actually for properly gathering a text from an index (since I really just want a drive letter). I will make this code more readable and understandable.